woocommerce/woocommerce-ios
# [WIP] Testing: Add UI tests for orders

## Changes

This PR adds three UI tests for orders:

* Fulfill an order
* Add an order note
* Add order tracking

This automates most of the Orders smoke test cases we use for beta testing, ensuring that user interactions (text input, button taps) work as expected and that the overall user flows work.

Note that this does **not** automate the smoke test cases for order push notifications (I'm still exploring how feasible this is to automate right now). Also, the focus here is on user flows/interactions and not verifying the order data on each screen.

_In Progress: Investigating CI failure on iPad._

## Testing

* Confirm the tests pass on CI (trigger the optional UI tests on this PR).
* Optionally, run the UI tests locally and confirm they pass. A quick way to run the UI tests locally:
  * In the simulator, disable the "Connect Hardware Keyboard" setting (this can interfere with text input in the test).
  * Run `rake mocks` to start the mocks server.
  * Select the `UITests` test plan in Xcode (in the test navigator or under Product > Test Plan).
  * Run the tests (in the test navigator or under Product > Test).

## Review

Only one reviewer is needed, although input from the whole team about test coverage/confidence is welcome. When reviewing, please confirm:

* The changes to the app itself (new accessibility identifiers) look ok.
* The new tests pass.
* The new tests give confidence about these user flows (the assertions look good and you feel confident that they are testing the right things).

Apologies for the number of lines of code changed in this PR. A lot of this comes from the new mocks (added under `WooCommerce/WooCommerceUITests/Mocks/`) — these don't need a detailed review as long as the UI tests pass without WireMock warnings.

## Submitter checklist

Update release notes:
